Topic-4-Javascript-Desing-Patterns/1/1_A.js

```diff
@@ -0,0 +1,64 @@
+//RENAME FILE BEFORE COMMIT
+class Movie {
+
+    constructor (title,year,duration) {
+        this.title = title;
+        this.year = year;
+        this.duration = duration;
+    }
+    play() {
+        console.log("playing...");
+    }
+    pause() {
+        console.log("pausing...");
+    }
+    resume() {
+        console.log("resuming...");
+    }
+}
+
+class Actor {
+    constructor (name,age) {
+        this.name = name;
+        this.age = age;
+    }
+}
+
+class EventEmiter {
+    constructor () {
+        this.events = {};
+    }
+    on (eventName , callback) {
+        if ( this.events[eventName]) {
+            this.events[eventName].push(callback);
+        } else {
+            this.events[eventName] = [callback];
+        }        
+    }
+    emit (eventName) {
+        this.events[eventName].forEach(callback =>
+            callback());       
+    }
+    off (eventName , callback) {
+        let newEvents = this.events[eventName].filter(c => c !== callback);
+        this.events[eventName] = newEvents;
+    }
+    
+}
+const logHola = function () {
+    console.log("holasas");
+}
+let test = new EventEmiter;
+
+test.on("hola",logHola);
+
+test.off("hola",logHola);
+
+test.emit("hola");
+
+let movie1 = new Movie;
+movie1.title = "Madagascar";
```

Comment:
This is an antipattern, you should pass `title` and `year` as `Movie` constructor parameters